### DISCUSSION TITLE
OneToOne Example

### DISCUSSION THREAD

**Discussion by FloMaetschke** [mikro-orm/mikro-orm#1419] [Q&A] [answered]
I am looking for an example of a unidirectional one-to-one relationship and how to set up my entities in mikro-orm for this where the foreign-key is saved in the child object pointing to its parent.

> Assume this example, a book has only one cover page, and a cover page is only for one book.
> The book collection should not know about the cover page, instead i want to store the reference to the book in the cover page collection.

[Book]   1------1  [CoverPage]
```

@Entity({ collection: 'books' })
export class Book  {
  @PrimaryKey()
  _id!: ObjectId;

  @Property()
  text: string;

  @OneToOne()
  cover: CoverPage;
}

@Entity({ collection: 'coverPage' })
export class CoverPage{
  @PrimaryKey()
  _id!: ObjectId;

  @Property()
  title: string;

  // this collection has a refrence to the book _id field.. but how to?
}

```

**In my nestjs rest service i want to save the book containing the cover in one step.**

```
  async save(dto: Book): Promise<string> {
    const entity = this.bookRepository.create(dto);
    await this.bookRepository.persist(entity);
    await this.bookRepository.flush();
     return entity.id;
  }
```

The result in the book collection should be like this:

```
{
"_id": "exampleBookId12345",
"text": "The long story of an example is to write clearly."
}

```
The result in the coverPage collection should be like this:

```
{
"_id": "exampleCoverPageValueId",
"bookId": "exampleBookId12345",
"title": "Long Example Story"
}

```

Thx so much!

**Comment by B4nan** [comment, marked as answer]
Have you seen this page? 

https://mikro-orm.io/docs/relationships/#onetoone

```ts
@Entity()
export class Book  {
  // inverse side, points to the owner where the FK is stored
  @OneToOne(() => CoverPage, p => p.book)
  cover: CoverPage;
}

@Entity()
export class CoverPage{
  // owning side, where the FK is stored
  @OneToOne(() => Book)
  cover: Book;
}
```

Btw `em.persist()` is sync method, no need to await it.

**Reply by FloMaetschke** [reply]
Thank you!
